launch/launch/actions/declare_launch_argument.py

```diff
@@ -90,7 +93,10 @@ def __init__(
         else:
             self.__default_value = normalize_to_list_of_substitutions(default_value)
```

Comment:
@v-lopez it'd be good to flag inconsistencies between `choices` and the `default_value` as early as possible.

Reply:
Agreed, but I'm a little bit lost regarding the substitution handling.

Would a valid approach be:
1. Check if default_value is not an instance of `Substitution`
2. If it is not, raise an error if the `default_value` is not in `choices`

Reply:
Correct.